### PR TITLE
+ Pad subj numbers

### PR DESCRIPTION
**Pull request by kliegl**
+ prefix and pad subject numbers
+ question about violin plot (ERROR: MethodError: no method matching GLM.DensePredChol)

### CODE REVIEW

**Review by dmbates — APPROVED** [review]
Thanks for doing this.  The way you have created the branch and the pull request is indeed the approach we prefer.

I haven't been able to decide what is happening with the lines in the violin plot.  If I check out the current main branch there is not an error.  For me the package versions are
```julia
(EmbraceUncertainty) pkg> status
Project EmbraceUncertainty v0.1.0
Status `~/projects/EmbraceUncertainty/Project.toml`
  [cbdf2221] AlgebraOfGraphics v0.6.18
  [69666777] Arrow v2.7.2
  [6e4b80f9] BenchmarkTools v1.5.0
  [336ed68f] CSV v0.10.14
⌃ [13f3f980] CairoMakie v0.11.11
  [324d7699] CategoricalArrays v0.10.8
  [8be319e6] Chain v0.6.0
  [9a962f9c] DataAPI v1.16.0
  [75880514] DataFrameMacros v0.4.1
  [a93c6f00] DataFrames v1.6.1
  [31c24e10] Distributions v0.25.108
  [8f03c58b] Effects v1.2.0
  [da1fdf0e] FreqTables v0.4.6
  [38e38edf] GLM v1.9.0
  [ff71e718] MixedModels v4.23.1
  [b12ae82c] MixedModelsMakie v0.4.1
  [76087f3c] NLopt v1.0.2
  [2dfb63ee] PooledArrays v1.4.3
  [6f49c342] RCall v0.14.1
  [27983f2f] RectangularFullPacked v0.2.0
  [6c6a2e73] Scratch v1.2.1
  [5064a6a7] StandardizedPredictors v1.0.1
  [82ae8749] StatsAPI v1.7.0
  [2913bbd2] StatsBase v0.34.3
  [3eaba693] StatsModels v0.7.3
  [bd369af6] Tables v1.11.1
  [9d95f2ec] TypedTables v1.4.6
  [a5390f91] ZipFile v0.10.1
  [f43a241f] Downloads v1.6.0
  [37e2e46d] LinearAlgebra
  [d6f4376e] Markdown
  [44cfe95a] Pkg v1.10.0
  [9a3f8284] Random
Info Packages marked with ⌃ have new versions available and may be upgradable.
```
However, if I check out this branch I get the problem with the call to `linear(` as you do.

I can approve these changes (and commit them to the main branch if you wish me to do that) but we should revisit the issue of the call to `linear`.  It may end up unnecessary if we decide to use only the response times for words and separately model the probability of a correct response according to word or nonword.

**Comment by kliegl** [comment]
The status numbers are the same on my computer.

I will switch to your other branch "db/TidierPlots".  Perhaps I can come up with ggplot-version of the the violin plot. :)

**Comment by dmbates** [comment]
Something peculiar is happening in the formation of the `byitem` table in that it is creating the `wrdlen` column as a `PooledVector{Int8, Int32, Vector{Int32}` instead of just `Vector{Int8}`.  This is really dumb because it is a more complicated, indirect storage format, which takes up over 4 times as much space.  I haven't been able to work out how to avoid that.

**Comment by kliegl** [comment]
The violin plot shows up as expected when I execute this file in branch  "db/TidierPlots"

**Comment by kliegl** [comment]
Could it be related to replacing BenchmarkTools and Profile with Chairmarks?

(EmbraceUncertainty) pkg> up
    Updating registry at `~/.julia/registries/General`
    Updating git-repo `https://github.com/JuliaRegistries/General.git`
    Updating `~/Library/CloudStorage/GoogleDrive-reinhold.kliegl@gmail.com/My Drive/JuliaMixedModels/EmbraceUncertainty/Project.toml`
  [0ca39b1e] + Chairmarks v1.2.1
  [337ecbd1] + TidierPlots v0.7.5
    Updating `~/Library/CloudStorage/GoogleDrive-reinhold.kliegl@gmail.com/My Drive/JuliaMixedModels/EmbraceUncertainty/Manifest.toml`
  [6e4b80f9] - BenchmarkTools v1.5.0
  [0ca39b1e] + Chairmarks v1.2.1
  [caabdcdb] + Cleaner v1.1.0
  [fe2206b3] + TidierData v0.15.2
  [337ecbd1] + TidierPlots v0.7.5
  [9abbd945] - Profile

**Comment by dmbates** [comment]
I'm reasonably sure that it is not related to Chairmarks.  It seems to be related to the vector type created in the `byitem` table for `wrdlen` and I'm not sure what that should change between the other branches and this one.

I'll have to look at it in more detail if we decide that this is important.  Right now I have a meeting coming up and need to catch a bus to the campus.  I'll correspond with you later.

**Comment by kliegl** [comment]
Well, it was my fault. I had changed

`ldttrial = dataset(:ELP_ldt_trial)`

to

`ldttrial = DataFrame(dataset(:ELP_ldt_trial))`

The reason was that I wanted to avoid the `DataFrame()` in the two `transform!()` in the next two chunks. 
Sorry about this.

File: largescaledesigned.qmd [hunk 1]
```diff
@@ -61,16 +61,27 @@ Initially we consider only the trial-level data.
...
 
-The two response variables are `acc` - the accuracy of the response - and `rt`, the response time in milliseconds.
 There is one trial-level covariate, `seq`, the sequence number of the trial within subj.
 Each subject participated in two sessions on different days, with 2000 trials recorded on the first day.
-The `s2` column, added to the data frame using `@transform!`, is a Boolean value indicating if the trial is in the second session.
+We add the `s2` column to the data frame using `@transform!`. The new variable `s2` is a Boolean value indicating if the trial is in the second session.
+
+```{julia}
+ldttrial =
+  transform!(ldttrial, :seq => (x -> x .> 2000) => :s2)
```
Comment (palday) [inline largescaledesigned.qmd:79]:
```suggestion
  transform!(ldttrial, :seq => ByRow(>(2000)) => :s2)
```

File: largescaledesigned.qmd [hunk 2]
```diff
@@ -61,16 +61,27 @@ Initially we consider only the trial-level data.
 ldttrial = dataset(:ELP_ldt_trial)
 ```
 
+Subject identifiers are coded in integers from 1 to 816. We prefer them as strings of the same length. 
+We prefix the subject number with 'S' and leftpad the number with zeros to three digits.
+
 ```{julia}
 ldttrial =
-  transform!(DataFrame(ldttrial), :seq => (x -> x .> 2000) => :s2)
+  transform!(DataFrame(ldttrial), :subj => (s -> string.('S', lpad.(s, 3, '0'))) => :subj)
```
Comment (palday) [inline largescaledesigned.qmd:69]:
```suggestion
  transform!(DataFrame(ldttrial), :subj => ByRow(s -> string('S', lpad(s, 3, '0'))) => :subj)
```

File: largescaledesigned.qmd [hunk 3]
```diff
@@ -579,6 +591,9 @@ and those associated with the subject are
 
 ```{julia}
 elpldtsubj = DataFrame(dataset(:ELP_ldt_subj))
+elpldtsubj =
+  transform!(elpldtsubj, :subj => (s -> string.('S', lpad.(s, 3, '0'))) => :subj)
```
Comment (palday) [inline largescaledesigned.qmd:595]:
```suggestion
  transform!(elpldtsubj, :subj => ByRow(s -> string('S', lpad(s, 3, '0'))) => :subj)
```

**Comment by palday** [comment]
@kliegl more specifically, `transform` and `transform!` are defined for DataFrames but not Arrow tables, so you need to put things into a DataFrame. The actual vectors themselves aren't copied -- the DataFrame just wraps the Arrow vectors. :smile:

**Comment by dmbates** [comment]
May I suggest that we just combine the two `transform!` calls to a single call, like
```julia
ldttrial = transform!(
  DataFrame(ldttrial),
  :subj =>
    (s -> string.('S', lpad.(s, maximum(ndigits, s), '0'))),
  :seq => ByRow(>(2000)) => :s2;
  renamecols=false,
)
describe(ldttrial)
```

This construction is a bit peculiar because `transform!` works in-place and generally you don't assign the result.  In this case it is equivalent to
```julia
ldtrial = DataFrame(ldttrial)
transform!(
  ldttrial,
  :subj => ...
)
```

I can make a commit with this change if that seems okay.

**Comment by kliegl** [comment]
Fine with me, of course. Thanks.

**Pull request merged**